Xplat/src/main/java/vazkii/botania/client/render/world/AstrolabePreviewHandler.java

```diff
@@ -53,10 +54,13 @@ public static void onWorldRenderLast(PoseStack ms, RenderBuffers buffers, Level
 	}
 
 	private static void renderPlayerLook(PoseStack ms, VertexConsumer buffer, Player player, ItemStack stack) {
-		List<BlockPos> coords = AstrolabeItem.getBlocksToPlace(stack, player);
-		if (AstrolabeItem.hasBlocks(stack, player, coords)) {
-			BlockState state = AstrolabeItem.getBlockState(stack);
-
+		Block blockToPlace = AstrolabeItem.getBlock(stack);
+		int size = AstrolabeItem.getSize(stack);
+		BlockPlaceContext ctx = AstrolabeItem.getBlockPlaceContext(player, player.swingingArm, blockToPlace);
+		List<BlockPos> coords = AstrolabeItem.getBlocksToPlace(ctx, size);
+		if (ctx != null && AstrolabeItem.hasBlocks(stack, player, coords.size(), blockToPlace)) {
+			BlockState stateForPlacement = blockToPlace.getStateForPlacement(ctx);
+			BlockState state = stateForPlacement != null ? stateForPlacement : blockToPlace.defaultBlockState();
```

Comment:
I'm pretty sure a null return from getStateForPlacement means the block says "I can't be here"?